go/pkg/client/cas.go

```diff
@@ -632,7 +632,11 @@ func (c *Client) DownloadActionOutputs(ctx context.Context, resPb *repb.ActionRe
 		}
 	}
 	for _, out := range copies {
-		if err := copyFile(execRoot, downloads[out.Digest].Path, out.Path); err != nil {
+		perm := c.RegularMode
+		if out.IsExecutable {
+			perm = c.ExecutableMode
+		}
+		if err := copyFile(execRoot, downloads[out.Digest].Path, out.Path, perm); err != nil {
```

Comment:
Given that we are pass RegularMode now as param to `copyFile()`, can you check here and make sure that `downloads[out.Digest]` is NOT a directory?

Reply:
done